climada/hazard/test/test_storm_europe.py

```diff
@@ -141,11 +144,59 @@ def test_generate_prob_storms(self):
         self.assertEqual(storms_prob.size, 60)
         self.assertTrue(np.allclose((1 / storms_prob.frequency).astype(int), 330))
         self.assertAlmostEqual(storms.frequency.sum(),
-                               storms_prob.frequency.sum())
+                                storms_prob.frequency.sum())
         self.assertEqual(np.count_nonzero(storms_prob.orig), 2)
         self.assertEqual(storms_prob.centroids.size, 3054)
         self.assertIsInstance(storms_prob.intensity,
                               sparse.csr.csr_matrix)
+        
+    def test_cosmoe_read(self):
+        """test reading from cosmo-e netcdf"""
+        haz = StormEurope()
+        haz.read_cosmoe_file(DATA_DIR.joinpath('storm_europe_cosmoe_forecast_vmax_testfile.nc'),
+                              run_date=dt.datetime(2018,1,1),
+                              event_date=dt.datetime(2018,1,3))
+        self.assertEqual(haz.tag.haz_type, 'WS')
+        self.assertEqual(haz.units, 'm/s')
+        self.assertEqual(haz.event_id.size, 21)
+        self.assertEqual(haz.date.size, 21)
+        self.assertEqual(dt.datetime.fromordinal(haz.date[0]).year, 2018)
+        self.assertEqual(dt.datetime.fromordinal(haz.date[0]).month, 1)
+        self.assertEqual(dt.datetime.fromordinal(haz.date[0]).day, 3)
+        self.assertEqual(haz.event_id[-1], 21)
+        self.assertEqual(haz.event_name[-1], '2018-01-03_ens21')
+        self.assertIsInstance(haz.intensity,
+                              sparse.csr.csr_matrix)
+        self.assertIsInstance(haz.fraction,
+                              sparse.csr.csr_matrix)
+        self.assertEqual(haz.intensity.shape, (21, 25))
+        self.assertAlmostEqual(haz.intensity.max(), 36.426735,places=3)
+        self.assertEqual(haz.fraction.shape, (21, 25))
+    
+    def test_icon_read(self):
```

Comment:
could you test for the appropriate warning if the hours are not 00 or 12?

(not sure one can test for logger output. else, would it make sense to actually raise an exception in the function?)
Or do the download functions that are called actually already take care of this?

Reply:
I added such a test, good idea.